------------------------------------------------------------
revno: 395
fixes bug(s): https://launchpad.net/bugs/752361
committer: Scott Moser <smoser@ubuntu.com>
branch nick: trunk
timestamp: Wed 2011-04-06 06:35:45 -0400
message:
  fix bug seeding grub dpkg config when devices are named xvdX (LP: #752361)
  
  With newer ubuntu kernels the xen block devices are named xvd[a-z][0-9]
  Ie, instead of /dev/sda1 the root device is /dev/xvda1.
=== modified file 'ChangeLog'
--- old/ChangeLog	2011-04-04 16:45:01 +0000
+++ new/ChangeLog	2011-04-06 10:35:45 +0000
@@ -7,6 +7,8 @@
    failure of 'blkid /dev/root' (LP: #726938)
  - convert dos formated files to unix for user-scripts, boothooks,
    and upstart jobs (LP: #744965)
+ - fix bug in seeding of grub dpkg configuration (LP: #752361) due
+   to renamed devices in newer (natty) kernels (/dev/sda1 -> /dev/xvda1)
 0.6.1:
  - fix bug in fixing permission on /var/log/cloud-init.log (LP: #704509)
  - improve comment strings in rsyslog file tools/21-cloudinit.conf

=== modified file 'cloudinit/CloudConfig/cc_grub_dpkg.py'
--- old/cloudinit/CloudConfig/cc_grub_dpkg.py	2010-09-11 19:19:18 +0000
+++ new/cloudinit/CloudConfig/cc_grub_dpkg.py	2011-04-06 10:35:45 +0000
@@ -31,7 +31,8 @@
         idevs_empty=util.get_cfg_option_str(cfg["grub-dpkg"],
             "grub-pc/install_devices_empty",None)
 
-    if os.path.exists("/dev/sda1") and not os.path.exists("/dev/sda"):
+    if (( os.path.exists("/dev/sda1") and not os.path.exists("/dev/sda") ) or
+        ( os.path.exists("/dev/xvda1") and not os.path.exists("/dev/xvda") )):
         if idevs == None: idevs=""
         if idevs_empty == None: idevs_empty="true"
     else:

